### PR TITLE
142790 queue message to create bespoke exchange set

### PR DESCRIPTION
**Pull request by chaital14503**
This PR consists changes of [PBI142790](https://dev.azure.com/ukhydro/Exchange%20Set%20Service/_sprints/taskboard/Exchange%20Set%20Service%20Team/Exchange%20Set%20Service/WP%20AM%20Bespoke%20ESS%20-%20Sprint%205?workitem=142790)

- Code changes for adding config properties in the queue "bess-fulfilment" when file size <= 700mb
- Unit test cases
- Excluded fluentvalidation.dll module from code coverage report.

### CODE REVIEW

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Helpers/AzureMessageQueueHelper.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+        // Create the queue if it doesn't already exist.
+        await queueClient.CreateIfNotExistsAsync();
+
+        // convert message to base64string
+        var messageBase64String = Convert.ToBase64String(System.Text.Encoding.UTF8.GetBytes(message));
+        // Send a message to the queue
+        await queueClient.SendMessageAsync(messageBase64String);
+
+        logger.LogInformation(EventIds.BessConfigPropertiesAddedInQueue.ToEventId(), "Added message in Queue:{queue}, QueueMessage: :{QueueMessage} and _X-Correlation-ID:{CorrelationId}", bessStorageConfiguration.QueueName, message, CommonHelper.CorrelationID);
```
Comment (harshal11869) [inline UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Helpers/AzureMessageQueueHelper.cs:33]:
```suggestion
        logger.LogInformation(EventIds.BessConfigPropertiesAddedInQueue.ToEventId(), "Added message in Queue:{queue}, QueueMessage :{QueueMessage} and _X-Correlation-ID:{CorrelationId}", bessStorageConfiguration.QueueName, message, CommonHelper.CorrelationID);
```

**Review by harshal11869 — CHANGES_REQUESTED** [review]

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Helpers/AzureMessageQueueHelper.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+namespace UKHO.PeriodicOutputService.Common.Helpers;
+
+[ExcludeFromCodeCoverage]
+public class AzureMessageQueueHelper : IAzureMessageQueueHelper
+{
+    private readonly ILogger<AzureMessageQueueHelper> logger;
+    private readonly BessStorageConfiguration bessStorageConfiguration;
+    public AzureMessageQueueHelper(ILogger<AzureMessageQueueHelper> logger, IOptions<BessStorageConfiguration> bessStorageConfiguration)
+    {
+        this.logger = logger;
```
Comment (Nadeem14456) [inline UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Helpers/AzureMessageQueueHelper.cs:17]:
```suggestion
        this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
```

**Review by Nadeem14456 — CHANGES_REQUESTED** [review]

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Helpers/AzureMessageQueueHelper.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+using UKHO.PeriodicOutputService.Common.Configuration;
+using UKHO.PeriodicOutputService.Common.Logging;
+
+namespace UKHO.PeriodicOutputService.Common.Helpers;
+
+[ExcludeFromCodeCoverage]
+public class AzureMessageQueueHelper : IAzureMessageQueueHelper
+{
+    private readonly ILogger<AzureMessageQueueHelper> logger;
+    private readonly BessStorageConfiguration bessStorageConfiguration;
```
Comment (harshal11869) [inline UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Helpers/AzureMessageQueueHelper.cs:14]:
```suggestion
    private readonly IOptions<BessStorageConfiguration> bessStorageConfiguration;
    
```

**Review by harshal11869 — CHANGES_REQUESTED** [review]

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Helpers/AzureMessageQueueHelper.cs (hunk 1 above)
Comment (chaital14503) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Helpers/AzureMessageQueueHelper.cs]: Resolved. Thanks.

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Helpers/AzureMessageQueueHelper.cs (hunk 2 above)
Comment (chaital14503) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Helpers/AzureMessageQueueHelper.cs]: Resolved. Thanks.

**Review by shirinbt14926 — APPROVED** [review]
LGTM

**Review by Mayuresh-Gawde — APPROVED** [review]
LGTM from QA!!

File: UKHO.PeriodicOutputService/UKHO.BESS.ConfigurationService/Services/ConfigurationService.cs [hunk 4]
```diff
@@ -213,13 +226,30 @@ public bool CheckConfigFrequencyAndSaveQueueDetails(IList<BessConfig> bessConfig
...
+                        {
+                            logger.LogWarning(EventIds.BessSizeExceedsThreshold.ToEventId(), "Bespoke Exchange Set size {fileSizeInMb}MB which is more than the threshold :{BESSize}MB, Bespoke Exchange Set will not be created for file:{FileName} | _X-Correlation-ID : {CorrelationId}", Math.Round(fileSizeInMb, 2), BESSize, config.FileName, CommonHelper.CorrelationID);
+
+                            continue;
+                        }
+
+                        //--save details to message queue --
+                        IEnumerable<string> encCellNames = encCells.Select(i => i.Item1).ToList();
+
+                        Task<bool> success = azureBlobStorageService.SetConfigQueueMessageModelAndAddToQueue(config, encCellNames, totalFileSize);
```
Comment (DanOwenUKHO) [inline UKHO.PeriodicOutputService/UKHO.BESS.ConfigurationService/Services/ConfigurationService.cs:243]: This asynchronous method is not awaited/waited, should it be?

**Review by DanOwenUKHO — COMMENTED** [review]
1 question, but otherwise looks good

File: UKHO.PeriodicOutputService/UKHO.BESS.ConfigurationService/Services/ConfigurationService.cs (hunk 4 above)
Comment (Nadeem14456) [reply UKHO.PeriodicOutputService/UKHO.BESS.ConfigurationService/Services/ConfigurationService.cs]: resolved.

**Review by DanOwenUKHO — APPROVED** [review]

**Pull request merged**